src/main.js

```diff
@@ -18,85 +20,113 @@ if (!org && org.length === 0) {
 
 const branchName = branchNameInput.replace('refs/heads/', '').replace(/[^a-zA-Z0-9-]/g, '-');
 const branchPattern = `-${branchName}.`;
-const octokit = github.getOctokit(token);
-
-async function getListOfPackages() {
-  let hasMorePackages = true;
-  let packagesToDelete = [];
-  let page = 1;
-  const maxResultsPerPage = 50;
-  core.info(`Gathering list of packages with '${branchPattern}' in the name or tag to delete...`);
-
-  while (hasMorePackages) {
-    const response = await octokit.rest.packages.getAllPackageVersionsForPackageOwnedByOrg({
-      org: org,
-      package_type: packageType,
-      package_name: packageName,
-      per_page: maxResultsPerPage,
-      page: page
-    });
-
-    if (response.status == 200) {
-      if (response.data) {
-        if (response.data.length < maxResultsPerPage) {
-          hasMorePackages = false;
-        } else {
-          page += 1;
-        }
+const graphqlWithAuth = graphql.defaults({
+  headers: {
+    authorization: `token ${token}`
+  }
+});
 
-        for (let index = 0; index < response.data.length; index++) {
-          const package = response.data[index];
-          if (package.name.indexOf(branchPattern) > -1) {
-            packagesToDelete.push({
-              name: package.name,
-              id: package.id
-            });
-          } else {
-            core.info(`Package ${package.name} does not meet the pattern and will not be deleted`);
+async function getAllPackagesForRepo() {
+  core.info('Querying for all of the packages in the repo.\n');
+  const query = `
+  query {
+    repository(owner: "${org}", name: "${repo}") {
+      packages(packageType: ${packageType.toUpperCase()}, first: 100){
+        totalCount
+        nodes {
+          name
+          id
+          versions(last: 100) {
```

Comment:
What if there are more than 100?

I can imagine this action running for the first time on a repository that has accumulated a long history of pre-release packages.

Reply:
You know, I didn't think of that. I thought the last 100 would be safe, but really I just wanted to be lazy about pagination because it's a little interesting with graphql. You bring up a likely scenario, though, so I'll stop being lazy and implement paging.